### DISCUSSION TITLE
Open AI chat history and model training

### DISCUSSION THREAD

**Discussion by SarahAmagno** [didalgolab/ai-intellij-plugin#18] [Q&A] [answered]
Hi :) 
I have a question regarding the OpenAI data controls: when using ChatGPT via web interface or app, I can turn chat history and model training on or off. Is there a similar option for this plugin? Or is the chat history always/never activated per default?
Regards, Sarah

**Comment by didalgolab** [comment, marked as answer]
Hi @SarahAmagno, I'm unsure if you are asking about using the plugin via API interfaces (with an API key) or if you are referring to the "Online ChatGPT" tab. If you are asking about usage via API, then currently the plugin doesn't store any chat history locally, and also OpenAI never trains models on your API-sent data. If chat history functionality will be added to the plugin, then an option to disable it will also be introduced.

If, on the other hand, you are asking about the "Online ChatGPT" interface, then the settings are exactly as you configure them for your ChatGPT account - in general, OpenAI may train on your data unless you opt out or have a ChatGPT Teams account.

**Reply by SarahAmagno** [reply]
Hi, I'm sorry I missed the fact that I can only activate the plugin via API key, there of course the data handling is different.
